utils/get_abstracts.py

```diff
@@ -0,0 +1,84 @@
+import synapseclient
+from synapseclient import Table
+import argparse
+import pandas as pd
+import requests
+import json
+
+
+### Login to Synapse ###
+def login():
+
+    syn = synapseclient.Synapse()
+    syn.login()
+
+    return syn
+
+
+def get_args():
+
+    parser = argparse.ArgumentParser(
+        description='Get synapse publications table id')
+    parser.add_argument('table_id',
+                        type=str,
+                        help='Synapse table id to upload manifest to.')
+
+    return parser.parse_args()
+
+
+def get_df(syn, publications_table_id):
+
+    pubs_query = (f"SELECT pubMedId, abstract FROM {publications_table_id}")
+    pubs_df = syn.tableQuery(pubs_query).asDataFrame().fillna("")
+
+    pmid_list = pubs_df['pubMedId'].tolist()
+
+    return pubs_df
+
+
+def get_pmids(pubs_df):
+
+    # Convert column of pubmedIds to a list
+    pmid_list = pubs_df['pubMedId'].tolist()
+
+    return (pmid_list)
+
+
+def get_abstracts(pmid_list, pubs_df):
+
+    for pmid in pmid_list:
+
+        endpoint = f"https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/pubtator?pmids={pmid}&concepts=none"
```

Comment:
Per PubTator:

> In order not to overload the PubTator server, we require that users post no more than three requests per second.

Since you're doing one call per PMID (rather than one bulk search), I recommend adding a small stop after each request with `sleep()`:

```
from time import sleep
...

for pmid in pmid_list:
   endpoint = f"https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/pubtator?pmids={pmid}&concepts=none"
   ...
   sleep(0.1)
```

Reply:
Thank you! I was trying to figure out how to do a bulk search, but from what I gathered, pubtator only seems to allow one pmid at a time, whereas if I used biocjson or biocsxml, I could do a bulk search, so I just went with the loop. I didn't test it yet with the whole list, so great to know! I will add the sleep!

Reply:
You should be able to bulk search with their pubtator format as well; results are delimited with two newlines, e.g.

```
endpoint = f"https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/pubtator?pmids={','.join(str(pmid) for pmid in pmid_list)}&concepts=none"
response = requests.get(endpoint).text
results = response.strip().split("\n\n")
for res in results:
   ...
```

Either way, so long as we're not sending too many requests, any approach you decide to take is fine!